### PR TITLE
CI: Upgrade/change dependencies to avoid deprecation warnings

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

**Review by tronical — APPROVED** [review]

**Comment by ogoffart** [comment]
Meh, running the test fails on macos because it doen't find Qt anymore.  What could have changed?

**Comment by tronical** [comment]
Weird, the path looks the same and marches dyld_framework_path. Or is there a different I missed?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -60,15 +60,9 @@ jobs:
...
-      with:
-          command: build
-          args: --verbose --all-features --workspace ${{ matrix.extra_args }} --exclude test-driver-cpp --exclude mcu-board-support --exclude printerdemo_mcu --exclude carousel  # mcu backend requires nightly
+      run: cargo build --verbose --all-features --workspace ${{ matrix.extra_args }} --exclude test-driver-cpp --exclude mcu-board-support --exclude printerdemo_mcu --exclude carousel  # mcu backend requires nightly
     - name: Run tests
-      uses: actions-rs/cargo@v1
-      with:
-          command: test
-          args: --verbose --all-features --workspace ${{ matrix.extra_args }} --exclude test-driver-cpp --exclude mcu-board-support --exclude printerdemo_mcu --exclude carousel   # mcu backend requires nightly
+      run: cargo test --verbose --all-features --workspace ${{ matrix.extra_args }} --exclude test-driver-cpp --exclude mcu-board-support --exclude printerdemo_mcu --exclude carousel   # mcu backend requires nightly
```
Comment (ogoffart) [inline .github/workflows/ci.yaml:65]: I'm going to try to revert just this part

```suggestion
      uses: actions-rs/cargo@v1
      with:
          command: test
          args: --verbose --all-features --workspace ${{ matrix.extra_args }} --exclude test-driver-cpp --exclude mcu-board-support --exclude printerdemo_mcu --exclude carousel   # mcu backend requires nightly
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
```
Comment (ogoffart) [inline .github/workflows/ci.yaml:36]:
```suggestion
    - uses: actions/checkout@v2
```

**Pull request merged**